src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -82,11 +82,20 @@ public static class ExpressionMetadataProvider
 
             object modelAccessor(object container)
             {
-                var compiledExpression = CachedExpressionCompiler.Process(expression);
-                Debug.Assert(compiledExpression != null);
+                var model = (TModel)container;
+                var cachedFunc = CachedExpressionCompiler.Process(expression);
+                if (cachedFunc != null)
+                {
+                    return cachedFunc(model);
+                }
+
+                // CachedExpressionCompiler returns null if the expression cannot be cached. In this event, fall back to 
```

Comment:
This comment is incorrect. For example `CompileFromConstLookup(...)` does not currently cache the expression.

Reply:
Removed